Remove all clippy warnings in tests

gbogard opened the pull request:
Hi 🙂 
This PR aims at removing the warnings emitted by clippy when running it against the test modules, using `cargo clippy --tests`. There were a number of usless uses of `format!` and `vec!` allocations in the tests.

```
error: useless use of `format!`
  --> tests/integration_test.rs:44:29
   |
44 |         assert!(!r.allowed(&format!("https://www.reddit.com/login")));
   |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider using `.to_string()`: `"https://www.reddit.com/login".to_string()`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_format
   = note: `-D clippy::useless-format` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::useless_format)]`
   ```
   
These warnings are not shown in the VS Code UI by default, because the default check command is `cargo check`, but they start to appear once we change the check command to the more advanced linter `clippy`, like so

```js
// In settings.json
{"rust-analyzer.check.command": "clippy"}
```

I think it might also be a good idea to add a `clippy` step to the CI workflow, but I haven't included it in this PR.

Cheers 🙂

Commit:
Remove all clippy warnings in tests

Smerity commented:
As these are in test which is seldom run I'm not too concerned with wayward `format!` / `vec!` allocations but you're indeed correct - best to make the `clippy` gods happy! lol

I'll avoid doing these in the future as your replacement is indeed cleaner and more efficient.

The only hold is that there are a few minor lines of issue when running `cargo fmt -- --check`.
If you're able to fix those I'll merge it in!

Also, if you know how to add the `clippy` step to CI feel welcome :)

Thanks for the improvements!

Commit:
CI: added a clippy job

gbogard commented:
Thanks for your kind feedback :) 

I have updated the PR to fix the formatting issues, and added a `clippy` job to the CI.
It should work as expected. I know because I have accidentally opened a testing PR with intentional clippy warnings on your repository, thinking I was opening it on my forked repository 😅 Anyway, sorry about that, #10 is closed, and this one should be good to go.

Smerity commented:
At least opening a PR into the wrong fork is entirely not a problem, compared to me once deleting the production database instead of the testing database on prod !

This PR is a improvement both for this repo and any future Rust repos I'll make.
Thanks for both cleaning up my own Clippy catastrophe and preventing future Clippy catastrophes too :)
